Support reading the Julia version from a `.tool-versions` file (#392)
julia-actions/setup-julia: README.md

    # Default: '1'
    version: '1'

    # Read the Julia version from a .tool-versions file.
    #

>     # This is useful for repositories that already pin Julia with asdf or mise.

Reviewer: ```suggestion
    # This is useful for repositories that already pin Julia with asdf or mise-en-place.
```

    # The file must contain a Julia entry, for example:

>     #     julia 1.10

Reviewer: ```suggestion
    #     julia 1.10.11
```

    #

>     # This input cannot be used together with version.

Reviewer: ```suggestion
    # The `version-file` input cannot be used together with the `version` input.
```

>
> It is strongly recommended to wrap versions in quotes. Otherwise, the YAML parser used by GitHub Actions parses certain versions as numbers which causes the wrong version to be selected. For example, `1.0` may be parsed as `1`.

Alternatively, set `version-file` to a `.tool-versions` file containing a Julia entry:

```yaml
steps:
- uses: actions/checkout@v4

> - uses: julia-actions/setup-julia@v2

Reviewer: ```suggestion
- uses: julia-actions/setup-julia@v3
```

  with:
    version-file: '.tool-versions'
```

> ```text

Reviewer: What is this? Is it an example `.tool-versions` file? You need some brief text here explaining.

> julia 1.10

Reviewer: ```suggestion
julia 1.10.11
```

```

> Only `.tool-versions` files are supported. The Julia entry must specify major, minor, and patch versions. The `version` and `version-file` inputs cannot be used together.

Reviewer: > Only `.tool-versions` files are supported.

This seems redundant. What else would we support? Can you remove this text?

>
> It is strongly recommended to wrap versions in quotes. Otherwise, the YAML parser used by GitHub Actions parses certain versions as numbers which causes the wrong version to be selected. For example, `1.0` may be parsed as `1`.

Alternatively, set `version-file` to a `.tool-versions` file containing a Julia entry:

```yaml
steps:

> - uses: actions/checkout@v4

Reviewer: ```suggestion
- uses: actions/checkout@v6
```
